jetty-core/jetty-ee/src/main/java/org/eclipse/jetty/ee/WebappProtectedClasses.java

```diff
@@ -0,0 +1,129 @@
+package org.eclipse.jetty.ee;
+
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.util.Attributes;
+import org.eclipse.jetty.util.ClassMatcher;
+import org.eclipse.jetty.util.component.Environment;
+
+public class WebappProtectedClasses
```

Comment:
Probably need javadoc for this class that explains its purpose, and clarifies System, Environment and webapp roles.